broker-daemon/state-machines/plugins/state-machine-events.js

```diff
@@ -0,0 +1,69 @@
+const EventEmitter = require('events')
+
+const StateMachinePlugin = require('./abstract')
+
+/**
+ * Events that should trigger the cleanup of all event handlers created
+ * by the StateMachineEvents plugin
+ *
+ * @constant
+ * @type {Array<String>}
+ * @default
+ */
+const EXIT_EVENTS = Object.freeze(['cancel', 'complete', 'reject'])
+
+/**
+ * @class Add event emittion to a state machine
+ */
+class StateMachineEvents extends StateMachinePlugin {
+  /**
+   * Setup for configuration of StateMachineEvents plugin
+   * @returns {StateMachineEvents}
+   */
+  constructor () {
+    super()
+
+    this.eventHandler = new EventEmitter()
+  }
+
+  /**
+   * Adds an event emition observer to all transition calls to allow for event passing
+   * to external workers (BlockOrderWorker)
+   */
+  get observers () {
+    const plugin = this
+
+    return {
+      /**
+       * @param {Object} lifecycle
+       * @returns {void}
+       */
+      onAfterTransition: function (lifecycle) {
+        // We check to see if an order exists, if it does then we are dealing with an
+        // OSM, or else we use the FSM
+        const blockOrderId = this.order ? this.order.blockOrderId : this.fill.blockOrderId
+
+        plugin.eventHandler.emit(lifecycle.transition, blockOrderId)
+
+        // Remove all events from the event emitter if we have either cancelled,
+        // completed, or rejected an order
+        if (EXIT_EVENTS.includes(lifecycle.transition)) {
+          plugin.eventHandler.removeAllListeners()
+        }
+      }
+    }
+  }
+
+  /**
+   * Alias methods on a StateMachine for event handling
+   */
+  get methods () {
+    const { eventHandler } = this
+
+    return {
+      once: (type, cb) => eventHandler.once(type, cb)
```

Comment:
why expose only `once`? why not provide the full API? e.g. `on`, `once`, `removeListener`, etc

Reply:
@treygriffith I've omitted those other events because they are not used

Reply:
Ok, let's punt for now